### ISSUE TITLE
Laravel PHPUnit Error Event Dispatacher

### ISSUE THREAD

**Issue by kontrollanten** [tymondesigns/jwt-auth#402]
When I use expectsEvents function i Laravel PHPUnit the jwt auth middleware throws an error.

This is the test:

```
    public function testPutCarts()
    {
        $cart = factory(App\Cart::class)->create();
        $this->expectsEvents(App\Providers\Events\CartStatusWasUpdated::class);
        $response = $this->call(
            'PUT', 
            $this->endpoint . '/' . $cart->id, 
            ['status' => $status],
            [],
            [],
            $headers);
    }
```
 And this is the response:

```
ErrorException: Argument 2 passed to Tymon\JWTAuth\Middleware\BaseMiddleware::__construct() must be an instance of Illuminate\Events\Dispatcher, instance of Mockery_0_Illuminate_Contracts_Events_Dispatcher given

vendor/tymon/jwt-auth/src/Middleware/BaseMiddleware.php:33
```

This issue seems like a similar https://github.com/cartalyst/sentinel/issues/166 which they solved in this commit https://github.com/cartalyst/sentinel/commit/3f2e7e89c5f85bd755b38875f6810bcfb0bf015c

**Comment by tdhsmith** [comment]
Ahh yes it looks like `expectsEvents` [mocks the contract](Illuminate\Contracts\Events\Dispatcher) and we're [requiring the implementer](https://github.com/tymondesigns/jwt-auth/blob/master/src/Middleware/BaseMiddleware.php#L33). I'll double-check if there's any important discrepancies or side effects between the two later today and see if I can get a simpler test case going.

**Comment by kontrollanten** [comment]
I couldn't contribute to the repo so I've attached a diff that uses the contracts instead. It solves my problem and it works with the phpunit tests :-) 

[#402.diff](https://github.com/tymondesigns/jwt-auth/files/101628/402.txt)

**Comment by tymondesigns** [comment]
merged and bundled with `0.5.7`
